client/client.go

```diff
@@ -49,17 +50,14 @@ type Client struct {
 	Transport http.RoundTripper
 
 	authed bool
-	inited bool
-	initMu sync.Mutex
+	inited int32
 	authMu sync.Mutex
 	auth   *auth.Request
 }
 
-func (c *Client) init() error {
-	c.initMu.Lock()
-	defer c.initMu.Unlock()
-	if c.inited {
-		return nil
+func (c *Client) init() {
+	if !atomic.CompareAndSwapInt32(&c.inited, 0, 1) {
```

Comment:
I'm a bit lost here. Do you expect that a client can be used from several goroutines?

If yes, I think we should keep the mutex to avoid race condition on using a client that has not finished to initialize to make requests.
If no, I don't see why we bother to use `sync/atomic`.